### PR TITLE
feat: Create mobile actions component

### PR DESCRIPTION
**Pull request by sebpearce**
## Changes

This is a work in progress.

The goal of this PR is to get the API of the TitleBlockZen to a state where it can start being implemented across Performance, while I continue to refactor and add features to its internals:

image

The main feature added here is the "mobile actions drawer", which is a sliding drawer component fixed to the bottom of the viewport on tablet and mobile views:

image

### CODE REVIEW

File: draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    return primaryAction.menuItems
+      .filter(item => typeof item.action === "string")
+      .map(item => (
+        <MenuItem
+          action={item.action}
+          label={item.label}
+          icon={item.hasOwnProperty("icon") ? item.icon : undefined}
+        />
+      ))
+  }
```
Comment (caseysinclair) [inline draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx:76]: Nice, love a good `.filter().map(` chaining!

**Comment by sebpearce** [comment]
@ActuallyACat @mbylstra This is one of those "we need this merged, but there's more to come immediately after" situations. Would both of you be OK with a post-merge review or a proper review on a later PR to organise the finer details?

There are things in here that I wouldn't usually sign off on, but I'm asking to merge this now so that PEX can get started on the rollout with the new API here – I'll be cleaning up the questionable stuff before the cycle's out.

What I could do is ping DST on my next PR and make that the "full review" one. How does that sound?

File: draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx [hunk 2]
```diff
@@ -27,20 +37,47 @@ type Props = {
   sectionTitle?: string
   sectionTitleDescription?: string
   handleHamburgerClick?: (event: React.MouseEvent) => void
-  primaryActions?: PrimaryActions
-  secondaryActions?: SecondaryActions
+  primaryAction: PrimaryActionProps
```
Comment (caseysinclair) [inline draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx:40]: We may need to make the `primaryAction` and optional thing as some pages have no calls to action

**Comment by sebpearce** [comment]
That said, if there are any concerns about the API, feel free to mention them. Some of the potential future changes to it don't worry me as much, e.g. things that are `string` and need more functionality in the future could be modified to `string | React.ReactNode` when we need them without having to do a breaking change.

File: draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx (hunk 2 above)
Comment (sebpearce) [reply draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx:40]: Ah good catch. I'll do that now

Comment (sebpearce) [reply draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockZen.tsx]: ✅

File: draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx (hunk 1 above)
Comment (sebpearce) [reply draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx:76]: 😎

**Comment by caseysinclair** [comment]
> @ActuallyACat @mbylstra This is one of those "we need this merged even though it isn't finished" situations. Would both of you be OK with a post-merge review or a proper review on a later PR to organise the finer details?
> 
> There are things in here that I wouldn't usually sign off on (e.g. I need to refactor and tidy up structure and TS stuff), but I'm asking to merge this now so that PEX can get started on the rollout with the new API here – I'll be cleaning up the questionable stuff before the cycle's out.
> 
> What I could do is ping DST on my next PR and make that the "full review" one. How does that sound?

@ActuallyACat @mbylstra It would be great if we can approve merge this as soon as possible 🙏 .

**Comment by caseysinclair** [comment]
> That said, if there are any concerns about the API, feel free to mention them. Some of the potential future changes to it don't worry me as much, e.g. things that are `string` and need more functionality in the future could be modified to `string | React.ReactNode` when we need them without having to do a breaking change.

**Comment by sebpearce** [comment]
Ah changing `primaryAction` to be optional foiled me 😂  Fixing now.

image

**Comment by sebpearce** [comment]
This is interesting `yarn tslint` locally returns no errors, but I'm still getting them on Buildkite:

image

**Pull request merged**